v4/integrations/nrgin/nrgin.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package nrgin instruments https://github.com/gin-gonic/gin applications.
+//
+// Use this package to instrument inbound requests handled by a gin.Engine.
+// Call nrgin.Middleware to get a gin.HandlerFunc which can be added to your
+// application as a middleware:
+//
+//	router := gin.Default()
+//	// Add the nrgin middleware before other middlewares or routes:
+//	router.Use(nrgin.Middleware(app))
+//
+// Example: https://github.com/newrelic/go-agent/tree/master/v3/integrations/nrgin/example/main.go
```

Comment:
I am not going to mark down all of the v3->v4 changes anymore, but there are plenty of them. I  don't know why I kept it up this far. 🍭